genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/DirectoryManifest.java

```diff
@@ -256,27 +312,31 @@ public long getTotalSizeOfFiles() {
         @Override
         public FileVisitResult preVisitDirectory(final Path dir, final BasicFileAttributes attrs) throws IOException {
             final ManifestEntry entry = this.buildEntry(dir, attrs, true);
-            log.debug("Created manifest entry for directory {}", entry);
-            this.builder.put(entry.getPath(), entry);
-
-            // Temporary hack To mitigate an ongoing issue.
-            // Building manifests for deep application dependencies directory trees is putting a strain on the
-            // application and causing user errors.
-            // Until a proper fix is in place, skip the dependencies sub-trees.
-            if (JobConstants.DEPENDENCY_FILE_PATH_PREFIX.equals(entry.getName())) {
-                return FileVisitResult.SKIP_SUBTREE;
+            if (this.filter.includeDirectory(dir, attrs)) {
```

Comment:
is there a case where we "include" a directory but don't walk it?

Reply:
That's what i'm doing for the `dependencies` folder. In the manifest it looks like an empty folder, but i think it's more clear than skipping it entirely.